socket_linux.go

```diff
@@ -26,13 +31,26 @@ func NewSocket(addr *net.UDPAddr, recvBuf int, reuseport bool) (net.PacketConn,
 		return nil, err
 	}
 
-	sockaddr := unix.SockaddrInet4{
-		Port: addr.Port,
-	}
-	if copied := copy(sockaddr.Addr[:], addr.IP); copied != net.IPv4len {
-		panic("did not copy enough bytes of ip address")
+	var sa unix.Sockaddr
+	if domain == unix.AF_INET {
+		sockaddr := &unix.SockaddrInet4{
+			Port: addr.Port,
+		}
+		if copied := copy(sockaddr.Addr[:], addr.IP); copied != net.IPv4len {
```

Comment:
I don't think this is possible: `SockaddrInet4.Addr` is of type `[4]byte`, and `addr.IP` is of type `[]byte`. I'm pretty sure I need to use `copy`? We could probably skip the error checking though ...

However, given that I'm now using `To4()` above, I've changed this to also use `To4()` as the source.